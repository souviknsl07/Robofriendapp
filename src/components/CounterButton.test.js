import { render, shallow } from "enzyme";
import React from "react";
import CounterButton from "./CounterButton";

it("expect to render CounterButton component", () => {
  const mockColor = "red";
  expect(render(<CounterButton color={mockColor} />)).toMatchSnapshot();
});

it("correctly increments the counter", () => {
  const mockColor = "red";
  const wrapper = shallow(<CounterButton color={mockColor} />);
  wrapper.find('[id="counter"]').simulate("click");
  wrapper.find('[id="counter"]').simulate("click");
  expect(wrapper.state()).toEqual({ count: 2 });
  expect(wrapper.props().color).toEqual("red");
});
