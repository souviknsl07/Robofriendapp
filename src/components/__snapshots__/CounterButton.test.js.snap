// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CounterButton component 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "color": "red",
      "id": "counter",
    },
    "children": Array [
      Node {
        "data": "Count: 0",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "button",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "color": undefined,
      "id": undefined,
    },
    "x-attribsPrefix": Object {
      "color": undefined,
      "id": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
